### DISCUSSION TITLE
Why do tensors with different dimensions not work in PyTorch?

### DISCUSSION THREAD

**Discussion by Harsha-lohana** [mrdbourke/pytorch-deep-learning#988] [Q&A]
```
Tensor=torch.tensor([[[1,2,4],
                      [1,2,2],
                      [1,2,1]]])
```
This code is working but 
```
Tensor=torch.tensor([[[1],
                      [1,2],
                      [1,2,1]]])
```
this is not working there might be a chance to generate this type of tensor, is it??
or what's the reason behind working with the same length?

**Comment by Arcane-WD** [comment]
The torch.tensor function fails with the provided code because the nested lists have different lengths. PyTorch tensors require that all sub-arrays be of the same shape to form a regular grid, making the entire structure a valid tensor. This is similar in case with NumPy as well.

Here's a solution:
import torch

# Padding shorter lists with zeros
Tensor = torch.tensor([[[1, 0, 0],
                        [1, 2, 0],
                        [1, 2, 1]]])

print(Tensor)
